pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -48,15 +52,60 @@ def __init__(self):
         # method
         self.latent_to_symptom_delay =\
             pe.Parameters.instance().latent_to_sympt_delay
+        # Defining the length of the model time step (in days, can be a
+        # fraction of day as well).
         self.model_time_step = 1 / pe.Parameters.instance().time_steps_per_day
         self.delay = np.floor(self.latent_to_symptom_delay
                               / self.model_time_step)
 
+        # Infectiousness progression
+        # Instantiate parameters to be used in update infectiousness
+        infectious_profile = pe.Parameters.instance().infectiousness_prof
+        inf_prof_resolution = len(infectious_profile) - 1
+        infectious_period = pe.Parameters.instance().asympt_infect_period
+        # Extreme case where model time step would be too small
+        max_inf_steps = 2550
+        # Define number of time steps a person is infectious:
+        num_infectious_ts =\
+            int(np.ceil(infectious_period / self.model_time_step))
+        if num_infectious_ts >= max_inf_steps:
+            raise ValueError('Number of infect timesteps exceeds limit')
+        # Initialisation
+        infectious_profile[inf_prof_resolution] = 0
+        infectiousness_prog = np.zeros(max_inf_steps)
+        sum_for_scaling = 0
+        # Fill infectiousness progression array
+        for i in range(num_infectious_ts):
+            t = (((i * self.model_time_step) / infectious_period)
+                 * inf_prof_resolution)
+            # Infectiousness value associated to infectiousness profile:
+            associated_inf_value = int(np.floor(t))
+            t -= associated_inf_value
+            if associated_inf_value < inf_prof_resolution:
+                infectiousness_prog[i] =\
+                    (infectious_profile[associated_inf_value] * (1 - t)
+                     + infectious_profile[associated_inf_value + 1] * t)
+                sum_for_scaling += infectiousness_prog[i]
+            else:  # limit case where we define infectiousness to 0
+                infectiousness_prog[i] =\
+                    infectious_profile[inf_prof_resolution]
+                sum_for_scaling += infectiousness_prog[i]
+        # Scaling
+        scaling_param = sum_for_scaling / num_infectious_ts
+        # Removes artificial case of division by 0, would only happen if every
```

Comment:
```suggestion
        # Avoids artificial case of division by 0, would only happen if every
```
I think this better describes what you are doing?